web/client/codechecker_client/cmd_line_client.py

```diff
@@ -771,9 +771,11 @@ def handle_diff_results(args):
 
     context = webserver_context.get_context()
 
+    client = None
+
     file_cache: Dict[int, File] = {}
 
-    def cached_report_file_lookup(file_id):
+    def cached_report_file_lookup(file_id: int, file_path: str) -> File:
```

Comment:
If I understand correctly, the goal of this second parameter is that `to_report()` function of the report converter can fill in the file path and content fields with the help of this function. I'd prefer the previous solution where `convert_report_data_to_report()` fills these fields for two design reasons:

- Now the command line client calls report converter with passing this function as a callback (the second parameter of `to_report()` is not even optional). This makes report converter dependent on CodeChecker. The design goal of any tool under `tools` directory is that they don't depend on CodeChecker.
- This function can be given a file ID and file path that are independent from each other.

Reply:
Please, add a short description about the purpose of this second parameter and then we can merge this PR. Thank you.

Reply:
Updated.